test/README.md

```diff
@@ -14,21 +14,35 @@ CONNECTION_PARAMETERS = {
 }
 ```
 
-### Running a single test
-
-Assuming that all dependencies are installed, running a single test is as simple as:
-`python -m pytest test/integ/test_connection.py::test_basic`.
-
 ### Running a suite of tests
 
-We use `tox` to run test suites and other utilities.
+We use `tox` version 4 to run test suites and other utilities.
 
 To run the most important tests, execute:
 
 ```shell
 tox -e "fix_lint,py37{,-pandas,-sso}"
 ```
 
+**NOTE** Some integration tests may be sensitive to the cloud provider of the
+account that the test suite connects to.  By default, when testing locally,
+the "provider" is called `dev`, but you may see some test failures with this
+provider (which isn't really a provider).  To eliminate this false failure,
+set the environment variable `cloud_provider` to one of `aws`, `gcp`, or
+`azure` as appropriate for the account you're running integration tests
+against.  This is handled correctly in CI so should only affect local
+developers.  In the future, we'll try to make this automatic by querying the
+account after the connection is made.
+
+### Running a single test
+
+Enter the tox environment you want (e.g. `py38`) and run `pytest` from there:
+
+```shell
+source .tox/py38/bin/activate
```

Comment:
So `source` is a bashism, while most things support bash syntax [dot](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot) should work on all modern shells (`dash` is the only one that comes to my mind that is a strict sh compatible shell).
```suggestion
. .tox/py38/bin/activate
```